AppController/djinn.rb

```diff
@@ -1729,22 +1719,20 @@ def restart_versions(versions_to_restart)
 
     Djinn.log_info("Restarting versions: #{versions_to_restart}.")
     # Self needs to update source code/cache.
-    if my_node.is_load_balancer?
-      versions_to_restart.each{ |version_key|
-        begin
-          HelperFunctions.parse_static_data(version_key, true)
-        rescue => except
-          except_trace = except.backtrace.join("\n")
-          Djinn.log_debug("restart_versions: parse_static_data exception" \
-            " from #{version_key}: #{except_trace}.")
-          # This specific exception may be a JSON parse error.
-          error_msg = "ERROR: Unable to parse app.yaml file for " +
-                      "#{version_key}. Exception of #{except.class} with " +
-                      "message #{except.message}"
-          place_error_app(version_key, error_msg)
-        end
-      }
-    end
+    versions_to_restart.each{ |version_key|
```

Comment:
Why do machines that aren't load balancers need to parse static data?

Reply:
The function is called only by the shadow, so the check is redundant: only shadow (who is a LB) will execute it.

Reply:
Oh, interesting. Does this even need to be run on the shadow?

Reply:
yeah only shadow can remove appservers.